front50-s3/src/main/java/com/netflix/spinnaker/front50/config/S3Config.java

```diff
@@ -0,0 +1,81 @@
+package com.netflix.spinnaker.front50.config;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.awsobjectmapper.AmazonObjectMapper;
+import com.netflix.spinnaker.clouddriver.aws.bastion.BastionConfig;
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonClientProvider;
+import com.netflix.spinnaker.front50.model.*;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.web.client.RestTemplate;
+import rx.schedulers.Schedulers;
+
+import java.util.concurrent.Executors;
+
+@Configuration
+@Import(BastionConfig.class)
+public class S3Config {
+  @Value("${spinnaker.s3.bucket}")
+  private String bucket;
+
+  @Value("${spinnaker.s3.rootFolder}")
+  private String rootFolder;
+
+  @Bean
+  public AmazonClientProvider amazonClientProvider() {
+    return new AmazonClientProvider();
+  }
+
+  @Bean
+  public AmazonS3 awsS3Client(AWSCredentialsProvider awsCredentialsProvider) {
+    return new AmazonS3Client(awsCredentialsProvider);
+  }
+
+  @Bean
+  @ConditionalOnMissingBean(RestTemplate.class)
+  public RestTemplate restTemplate() {
+    return new RestTemplate();
+  }
+
+  @Bean
+  public ObjectMapper amazonObjectMapper() {
+    return new AmazonObjectMapper();
+  }
+
+  @Bean
+  @ConditionalOnExpression("${spinnaker.s3.enabled:false}")
```

Comment:
Make that
spinnaker:
  s3:
     enabled: ${global.spinnaker.s3.enabled:false}


On Tue, Mar 15, 2016 at 9:08 AM, Eric Wiseblatt <ewiseblatt@google.com>
wrote:

> I prefer spinnaker.s3 to scope it.
>
> I'd also add a corresponding entry to the front50.yml as well
> spinnaker:
>   s3:
>      enabled: ${global.spinnaker.s3.enabled}
>
> And add to spinnaker.yml (and default-spinnaker-local.yml)
> global:
>   spinnaker:
>      s3:
>        enabled: false
>
>
> On Mon, Mar 14, 2016 at 11:46 PM, Adam Jordens <notifications@github.com>
> wrote:
>
>> In
>> front50-s3/src/main/java/com/netflix/spinnaker/front50/config/S3Config.java
>> <https://github.com/spinnaker/front50/pull/73#discussion_r56111975>:
>>
>> > +    return new AmazonS3Client(awsCredentialsProvider);
>> > +  }
>> > +
>> > +  @Bean
>> > +  @ConditionalOnMissingBean(RestTemplate.class)
>> > +  public RestTemplate restTemplate() {
>> > +    return new RestTemplate();
>> > +  }
>> > +
>> > +  @Bean
>> > +  public ObjectMapper amazonObjectMapper() {
>> > +    return new AmazonObjectMapper();
>> > +  }
>> > +
>> > +  @Bean
>> > +  @ConditionalOnExpression("${spinnaker.s3.enabled:false}")
>>
>> Any strong preference @cfieber <https://github.com/cfieber> on
>> spinnaker.s3 vs s3?
>>
>> Part of me is inclined to drop the spinnaker..
>>
>> I could also updated CassandraConfig and add support for both
>> cassandra.cluster and spinnaker.cassandra.cluster` values ... even
>> though I suspect in the long run that Cassandra will be removed in entirety.
>>
>> —
>> You are receiving this because you are on a team that was mentioned.
>> Reply to this email directly or view it on GitHub:
>> https://github.com/spinnaker/front50/pull/73/files#r56111975
>>
>
>